WalletWasabi/WabiSabi/Backend/Banning/CoinVerifier.cs

```diff
@@ -148,8 +149,8 @@ private bool CheckForFlags(ApiResponseItem response)
 		}
 
 		shouldBan = flagIds.Any(id => WabiSabiConfig.RiskFlags.Contains(id));
-
-		return shouldBan;
+		shouldRemove = !response.Report_info_section.Address_used;
```

Comment:
Quick fix could be this
```suggestion
		shouldRemove = shouldBan || !response.Report_info_section.Address_used;
```

Or make sure the record is updated properly by changing the property setters.